src/lib.rs

```diff
@@ -2,269 +2,186 @@
 //
 // SPDX-License-Identifier: MIT OR Apache-2.0
 
-/*! The library provide tools for handling compressed and archive files
-
-# Examples
-```no_run
-use compress_tools;
-
-let dir = tempfile::tempdir().unwrap();
-compress_tools::uncompress("tests/fixtures/tree.tar.gz", dir.path(), compress_tools::Kind::TarGZip).unwrap();
-```
-*/
-
-mod uncompress;
-
-pub use uncompress::{Error, Result};
-
-use std::path::Path;
-
-/// Type of compressed file or archive
-#[derive(Copy, Clone, Debug)]
-pub enum Kind {
-    TarGZip,
-    TarBZip2,
-    TarXz,
-    TarLZMA,
-    TarLZip,
-    Tar,
-    Zip,
-
-    GZip,
-    BZip2,
-    Xz,
-    LZip,
-    LZMA,
-}
-
-/// Uncompress a archive of known [Kind](Kind) pointed by source to target
-/// location for file
-pub fn uncompress<P1, P2>(source: P1, target: P2, kind: Kind) -> Result<()>
+pub mod error;
+pub mod ffi;
+pub mod util;
+
+use error as compress;
+use ffi::*;
+use std::{
+    ffi::{CStr, CString},
+    os::raw::c_void,
+};
+use util::*;
+
+fn init_archive<F>(mut f: F) -> compress::Result<()>
```

Comment:
I believe it's better to have public functions on the start of the file, and local helpers such as this one more the end.